src/jabs/ui/search_bar_widget.py

```diff
@@ -0,0 +1,194 @@
+from PySide6 import QtCore, QtWidgets
+
+from jabs.behavior_search import (
+    BehaviorSearchQuery,
+    LabelBehaviorSearchQuery,
+    PredictionBehaviorSearchQuery,
+    PredictionSearchKind,
+    SearchHit,
+    search_behaviors,
+)
+from jabs.project import Project
+
+
+class SearchBarWidget(QtWidgets.QWidget):
```

Comment:
not really relevant to this PR, but I'm regretting naming all my custom Qt Widgets *Widget. In Qt, things that extend the QWidget aren't named SomethingWidget.